tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -104,17 +106,102 @@ public function userListAllNotifications(string $user):void {
 		$this->featureContext->setResponse($response);
 	}
 
+	/**
+	 * @When user :user deletes all notifications
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws JsonException
+	 */
+	public function userDeletesAllNotifications(string $user):void {
+		$this->userListAllNotifications($user);
+		if (isset($this->featureContext->getJsonDecodedResponseBodyContent()->ocs->data)) {
+			$responseBody = $this->featureContext->getJsonDecodedResponseBodyContent()->ocs->data;
+			foreach ($responseBody as $value) {
+				// set notificationId
+				$this->notificationIds[] = $value->notification_id;
+			}
+		}
+		$this->featureContext->setResponse($this->userDeletesNotification($user));
+	}
+
+	/**
+	 * @When user :user deletes a notification related to resource :resource with subject :subject
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param string $subject
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws JsonException
+	 */
+	public function userDeletesNotificationOfResourceAndSubject(string $user, string $resource, string $subject):void {
+		$this->userListAllNotifications($user);
+		$this->filterResponseByNotificationSubjectAndResource($subject, $resource);
```

Comment:
I think this function returns some array of notification which is of no use. i sense this function just check whether the subject and resource is there in the response or not. May be make the function that returns boolean instead.

Reply:
I use returned array in my next step. This step doesn't need an array response so I didn't save it in a variable. `filterResponseByNotificationSubjectAndResource` can be used in a multiple way if we need notification id then we don't need an array but if we need the whole notification response then we need array value.